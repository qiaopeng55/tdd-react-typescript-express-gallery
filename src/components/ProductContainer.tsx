import { Typography } from "@material-ui/core";
import * as React from "react";
import { Col, Row } from "react-flexbox-grid";
import { Product } from "../model/model";
import { CardContainer, Divider } from "./StyledComponents";

const ProductContainer = ({ product }: { product: Product }) => (
  <Col lg={3} md={4} sm={6} xs={12}>
    <CardContainer>
      <div className="image">
        <img src={product.product_image} />
      </div>

      <Divider />
      <div className="text__area">
        <Typography variant="body1">
          {product.id} {product.product_name}
        </Typography>
        <Typography variant="caption">{product.description}</Typography>
        <Typography variant="subtitle2">{product.price}</Typography>
      </div>
    </CardContainer>
  </Col>
);

export default ProductContainer;
