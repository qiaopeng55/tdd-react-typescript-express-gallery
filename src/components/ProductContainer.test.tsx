import { shallow } from "enzyme";
import * as React from "react";
// import { MemoryRouter } from "react-router-dom";
import ProductContainer from "./ProductContainer";

const product = {
  id: 5,
  price: "$30.22",
  product_name: "Cyclobenzaprine Hydrochloride",
  description: "reinvent back-end deliverables",
  product_image: "http://dummyimage.com/324x341.bmp/dddddd/000000"
};

describe("ProductContainer", () => {
  const wrapper = shallow(<ProductContainer product={product} />);

  it("renders properly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
