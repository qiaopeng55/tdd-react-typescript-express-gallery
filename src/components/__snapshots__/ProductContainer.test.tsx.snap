// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductContainer renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductContainer
    product={
      Object {
        "description": "reinvent back-end deliverables",
        "id": 5,
        "price": "$30.22",
        "product_image": "http://dummyimage.com/324x341.bmp/dddddd/000000",
        "product_name": "Cyclobenzaprine Hydrochloride",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef>
        <div
          className="image"
        >
          <img
            src="http://dummyimage.com/324x341.bmp/dddddd/000000"
          />
        </div>
        <ForwardRef />
        <div
          className="text__area"
        >
          <WithStyles(Typography)
            variant="body1"
          >
            5
             
            Cyclobenzaprine Hydrochloride
          </WithStyles(Typography)>
          <WithStyles(Typography)
            variant="caption"
          >
            reinvent back-end deliverables
          </WithStyles(Typography)>
          <WithStyles(Typography)
            variant="subtitle2"
          >
            $30.22
          </WithStyles(Typography)>
        </div>
      </ForwardRef>,
      "lg": 3,
      "md": 4,
      "sm": 6,
      "xs": 12,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="image"
          >
            <img
              src="http://dummyimage.com/324x341.bmp/dddddd/000000"
            />
          </div>,
          <ForwardRef />,
          <div
            className="text__area"
          >
            <WithStyles(Typography)
              variant="body1"
            >
              5
               
              Cyclobenzaprine Hydrochloride
            </WithStyles(Typography)>
            <WithStyles(Typography)
              variant="caption"
            >
              reinvent back-end deliverables
            </WithStyles(Typography)>
            <WithStyles(Typography)
              variant="subtitle2"
            >
              $30.22
            </WithStyles(Typography)>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              src="http://dummyimage.com/324x341.bmp/dddddd/000000"
            />,
            "className": "image",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": "http://dummyimage.com/324x341.bmp/dddddd/000000",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-bxivhb",
              "isStatic": false,
              "rules": Array [
                "
  border: ",
                [Function],
                "px solid #eeeeee;
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-bxivhb",
            "target": "div",
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <WithStyles(Typography)
                variant="body1"
              >
                5
                 
                Cyclobenzaprine Hydrochloride
              </WithStyles(Typography)>,
              <WithStyles(Typography)
                variant="caption"
              >
                reinvent back-end deliverables
              </WithStyles(Typography)>,
              <WithStyles(Typography)
                variant="subtitle2"
              >
                $30.22
              </WithStyles(Typography)>,
            ],
            "className": "text__area",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  5,
                  " ",
                  "Cyclobenzaprine Hydrochloride",
                ],
                "variant": "body1",
              },
              "ref": null,
              "rendered": Array [
                5,
                " ",
                "Cyclobenzaprine Hydrochloride",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "reinvent back-end deliverables",
                "variant": "caption",
              },
              "ref": null,
              "rendered": "reinvent back-end deliverables",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "$30.22",
                "variant": "subtitle2",
              },
              "ref": null,
              "rendered": "$30.22",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  background: white;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-top: 15px;
  border: 1px solid #eeeeee;

  .image {
    display: flex;
    justify-content: center;
    align-items: center;
    margin: 10px;
    height: 200px;

    img {
      height: 100%;
    }
  }

  .text__area {
    margin: 15px;
  }
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef>
          <div
            className="image"
          >
            <img
              src="http://dummyimage.com/324x341.bmp/dddddd/000000"
            />
          </div>
          <ForwardRef />
          <div
            className="text__area"
          >
            <WithStyles(Typography)
              variant="body1"
            >
              5
               
              Cyclobenzaprine Hydrochloride
            </WithStyles(Typography)>
            <WithStyles(Typography)
              variant="caption"
            >
              reinvent back-end deliverables
            </WithStyles(Typography)>
            <WithStyles(Typography)
              variant="subtitle2"
            >
              $30.22
            </WithStyles(Typography)>
          </div>
        </ForwardRef>,
        "lg": 3,
        "md": 4,
        "sm": 6,
        "xs": 12,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <div
              className="image"
            >
              <img
                src="http://dummyimage.com/324x341.bmp/dddddd/000000"
              />
            </div>,
            <ForwardRef />,
            <div
              className="text__area"
            >
              <WithStyles(Typography)
                variant="body1"
              >
                5
                 
                Cyclobenzaprine Hydrochloride
              </WithStyles(Typography)>
              <WithStyles(Typography)
                variant="caption"
              >
                reinvent back-end deliverables
              </WithStyles(Typography)>
              <WithStyles(Typography)
                variant="subtitle2"
              >
                $30.22
              </WithStyles(Typography)>
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                src="http://dummyimage.com/324x341.bmp/dddddd/000000"
              />,
              "className": "image",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "src": "http://dummyimage.com/324x341.bmp/dddddd/000000",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-bxivhb",
                "isStatic": false,
                "rules": Array [
                  "
  border: ",
                  [Function],
                  "px solid #eeeeee;
",
                ],
              },
              "displayName": "styled.div",
              "render": [Function],
              "styledComponentId": "sc-bxivhb",
              "target": "div",
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <WithStyles(Typography)
                  variant="body1"
                >
                  5
                   
                  Cyclobenzaprine Hydrochloride
                </WithStyles(Typography)>,
                <WithStyles(Typography)
                  variant="caption"
                >
                  reinvent back-end deliverables
                </WithStyles(Typography)>,
                <WithStyles(Typography)
                  variant="subtitle2"
                >
                  $30.22
                </WithStyles(Typography)>,
              ],
              "className": "text__area",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    5,
                    " ",
                    "Cyclobenzaprine Hydrochloride",
                  ],
                  "variant": "body1",
                },
                "ref": null,
                "rendered": Array [
                  5,
                  " ",
                  "Cyclobenzaprine Hydrochloride",
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "reinvent back-end deliverables",
                  "variant": "caption",
                },
                "ref": null,
                "rendered": "reinvent back-end deliverables",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "$30.22",
                  "variant": "subtitle2",
                },
                "ref": null,
                "rendered": "$30.22",
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "rules": Array [
              "
  background: white;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-top: 15px;
  border: 1px solid #eeeeee;

  .image {
    display: flex;
    justify-content: center;
    align-items: center;
    margin: 10px;
    height: 200px;

    img {
      height: 100%;
    }
  }

  .text__area {
    margin: 15px;
  }
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
